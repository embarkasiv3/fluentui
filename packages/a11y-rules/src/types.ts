// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type PropValue = string | number | boolean | any;
export type Props = { [name: string]: PropValue };
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type AccessibilityBehavior = (props: Props) => any;

export interface Rule {
  stringify: () => string;
}

export interface Slot {
  name: string;
  props: Props[];
  expectAttribute: boolean;
  expectedAttribute: string;
  expectedValue: PropValue;
  description: string;
  hidden: boolean;
  afterEventData: Event;
  checkClick: boolean;
  checkSpaceKeyPressed: boolean;
  checkEnterKeyPressed: boolean;
  checkOnClickWasExecuted: boolean;
}
