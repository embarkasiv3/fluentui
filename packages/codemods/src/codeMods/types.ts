import { SourceFile } from 'ts-morph';

export interface CodeModResult {
  success?: boolean;
}

export interface Codemod<T = SourceFile> {
  /**
   * Each type of codemod can have multiple versions which work on different versions of its targeted package.
   * Must be valid semver.
   * TODO: Currently version is not being checked. Implement this fully or remove.
   */
  version?: string;
  /**
   * A string to help identify the codemod.
   */
  name: string;
  /**
   * The actual function that should be run on any given file.
   * TODO, is there a possibility of codemods that would need to execute over mutiple files?
   */
  run: (file: T) => CodeModResult;
  /**
   * If not enabled, then this mod will not be conisdered to run. Only enable it once it's ready
   * to be applied in real world scenarios
   */
  enabled?: boolean;
}
