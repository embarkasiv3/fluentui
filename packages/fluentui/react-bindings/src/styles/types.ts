import { Renderer } from '@fluentui/react-northstar-styles-renderer';
import { DebugData, ICSSInJSStyle, PropsWithVarsAndStyles, ThemeInput, ThemePrepared } from '@fluentui/styles';

import { Telemetry } from '../telemetry/types';

// Notice:
// This temporary lives here, will be remove once `animation` prop will be dropped
export type ComponentAnimationProp =
  | {
      name: string;
      delay?: string;
      direction?: string;
      duration?: string;
      fillMode?: string;
      iterationCount?: string;
      playState?: string;
      timingFunction?: string;
      keyframeParams?: object;
    }
  | string;

export type ComponentSlotClasses = Record<string, string>;

// Heads Up!
// Keep in sync with packages/react-proptypes/src/index.ts
export type ComponentDesignProp = {
  display?: ICSSInJSStyle['display'];
  position?: ICSSInJSStyle['position'];

  top?: ICSSInJSStyle['top'];
  bottom?: ICSSInJSStyle['bottom'];
  left?: ICSSInJSStyle['left'];
  right?: ICSSInJSStyle['right'];

  padding?: ICSSInJSStyle['padding'];
  paddingTop?: ICSSInJSStyle['paddingTop'];
  paddingRight?: ICSSInJSStyle['paddingRight'];
  paddingBottom?: ICSSInJSStyle['paddingBottom'];
  paddingLeft?: ICSSInJSStyle['paddingLeft'];

  margin?: ICSSInJSStyle['margin'];
  marginTop?: ICSSInJSStyle['marginTop'];
  marginRight?: ICSSInJSStyle['marginRight'];
  marginBottom?: ICSSInJSStyle['marginBottom'];
  marginLeft?: ICSSInJSStyle['marginLeft'];

  width?: ICSSInJSStyle['width'];
  height?: ICSSInJSStyle['height'];
  minWidth?: ICSSInJSStyle['minWidth'];
  maxWidth?: ICSSInJSStyle['maxWidth'];
  minHeight?: ICSSInJSStyle['minHeight'];
  maxHeight?: ICSSInJSStyle['maxHeight'];
};

export interface StylesContextPerformance {
  enableSanitizeCssPlugin: boolean;
  enableStylesCaching: boolean;
  enableVariablesCaching: boolean;
  enableBooleanVariablesCaching: boolean;
}

export type StylesContextPerformanceInput = Partial<StylesContextPerformance>;

export type StylesContextInputValue = {
  rtl?: boolean;
  disableAnimations?: boolean;
  performance?: StylesContextPerformanceInput;
  renderer?: Renderer;
  theme?: ThemeInput;
};

export type StylesContextValue = {
  rtl: boolean;
  disableAnimations: boolean;
  performance: StylesContextPerformance;
  renderer: Renderer;
  theme: ThemePrepared;
  telemetry?: Telemetry;
};

export type PrimitiveProps = Record<string, boolean | number | string | undefined>;

export type ResolveStylesOptions = StylesContextValue & {
  className?: string;
  allDisplayNames: string[];
  primaryDisplayName: string;
  componentProps: Record<string, any>;
  inlineStylesProps: PropsWithVarsAndStyles & { design?: ComponentDesignProp };
  rtl: boolean;
  telemetry?: Telemetry;
  saveDebug: (debug: DebugData | null) => void;
};
