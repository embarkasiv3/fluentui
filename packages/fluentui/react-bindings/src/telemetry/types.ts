type ComponentPerfStats = {
  instances: number;
  renders: number;

  msTotal: number;
  msMin: number;
  msMax: number;

  stylesRootCacheHits: number;
  stylesSlotsCacheHits: number;
};

export type UseTelemetryResult = {
  setStart: () => void;
  setEnd: () => void;
};

export class Telemetry {
  performance: Record<string, ComponentPerfStats>;
  enabled: boolean;

  constructor() {
    this.performance = {};
    this.enabled = true;
  }

  reset() {
    this.performance = {};
  }
}
