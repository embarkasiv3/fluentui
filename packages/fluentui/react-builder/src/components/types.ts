export type JSONTreeElement = {
  type: string;
  $$typeof?: 'Symbol(react.element)';
  displayName?: string;
  uuid: string | number;
  props?: {
    [key: string]: any;
    children?: (string | JSONTreeElement)[];
  };
};

export type DesignerMode = 'build' | 'design' | 'use';
