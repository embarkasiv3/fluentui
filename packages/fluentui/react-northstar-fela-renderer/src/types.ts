import { IRenderer } from 'fela';
import { ICSSInJSStyle } from '@fluentui/styles';

export type Renderer = IRenderer & {
  cache: Record<string, RendererChange>;
  selectorPrefix?: string;

  getNextRuleIdentifier: () => string;
  filterClassName: () => boolean;

  _emitChange: (change: RendererChange) => void;
  _renderStyleToClassNames(
    { _className, ...style }: ICSSInJSStyle & { _className: string },
    pseudo?: string,
    media?: string,
    support?: string,
  ): string;
};

export type RendererChange = {
  type: 'RULE' | 'KEYFRAME' | 'FONT' | 'STATIC' | 'CLEAR';
  className: string;
  selector: string;
  declaration: string;
  pseudo: string;
  media: string;
  support: string;
};
