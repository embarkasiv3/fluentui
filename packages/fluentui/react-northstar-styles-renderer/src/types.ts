import { ICSSInJSStyle, FontFace } from '@fluentui/styles';
import * as React from 'react';

export type RendererParam = {
  direction: 'ltr' | 'rtl';
  disableAnimations: boolean;
  displayName: string;
  sanitizeCss: boolean;
};

export type RendererRenderFont = (font: FontFace) => void;
export type RendererRenderGlobal = (styles: ICSSInJSStyle | string, selector?: string) => void;
export type RendererRenderRule = (styles: ICSSInJSStyle, param: RendererParam) => string;

export type Renderer = {
  renderGlobal: RendererRenderGlobal;
  renderFont: RendererRenderFont;
  renderRule: RendererRenderRule;

  Provider: React.FC;
};

export type CreateRenderer = (target?: Document) => Renderer;
