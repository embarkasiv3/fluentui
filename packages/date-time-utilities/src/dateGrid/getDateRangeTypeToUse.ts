import { DateRangeType, DayOfWeek } from '../dateValues/dateValues';

/**
 * Return corrected date range type, given `dateRangeType` and list of working days.
 * For non-contiguous working days and working week range type, returns general week range type.
 * For other cases returns input date range type.
 * @param dateRangeType - input type of range
 * @param workWeekDays - list of working days in a week
 */
export const getDateRangeTypeToUse = (
  dateRangeType: DateRangeType,
  workWeekDays: DayOfWeek[] | undefined,
): DateRangeType => {
  if (workWeekDays && dateRangeType === DateRangeType.WorkWeek) {
    const sortedWWDays = workWeekDays.slice().sort();
    let isContiguous = true;
    for (let i = 1; i < sortedWWDays.length; i++) {
      if (sortedWWDays[i] !== sortedWWDays[i - 1] + 1) {
        isContiguous = false;
        break;
      }
    }

    if (!isContiguous || workWeekDays.length === 0) {
      return DateRangeType.Week;
    }
  }

  return dateRangeType;
};
