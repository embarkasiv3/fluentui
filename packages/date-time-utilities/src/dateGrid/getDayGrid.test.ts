import { DateRangeType, DayOfWeek, FirstWeekOfYear } from '../dateValues/dateValues';
import { IDayGridOptions } from './dateGrid.types';
import * as DateGrid from './getDayGrid';

enum Months {
  Jan = 0,
  Feb = 1,
  Mar = 2,
  Apr = 3,
  May = 4,
  Jun = 5,
  Jul = 6,
  Aug = 7,
  Sep = 8,
  Oct = 9,
  Nov = 10,
  Dec = 11,
}

describe('Timezones', () => {
  it('should always be UTC', () => {
    expect(new Date().getTimezoneOffset()).toBe(0);
  });
});

describe('getDayGrid', () => {
  const defaultOptions: IDayGridOptions = {
    selectedDate: new Date(2016, Months.Apr, 1),
    navigatedDate: new Date(2016, Months.Apr, 1),
    firstDayOfWeek: DayOfWeek.Sunday,
    firstWeekOfYear: FirstWeekOfYear.FirstDay,
    dateRangeType: DateRangeType.Day,
  };

  it('returns matrix with days', () => {
    const result = DateGrid.getDayGrid(defaultOptions);
    expect(result).toMatchSnapshot();
  });
});
