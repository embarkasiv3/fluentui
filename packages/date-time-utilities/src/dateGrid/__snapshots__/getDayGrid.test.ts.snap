// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDayGrid returns matrix with days 1`] = `
Array [
  Array [
    Object {
      "date": "20",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Mar 20 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-20T00:00:00.000Z,
    },
    Object {
      "date": "21",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Mar 21 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-21T00:00:00.000Z,
    },
    Object {
      "date": "22",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Mar 22 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-22T00:00:00.000Z,
    },
    Object {
      "date": "23",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Mar 23 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-23T00:00:00.000Z,
    },
    Object {
      "date": "24",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Mar 24 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-24T00:00:00.000Z,
    },
    Object {
      "date": "25",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Fri Mar 25 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-25T00:00:00.000Z,
    },
    Object {
      "date": "26",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Mar 26 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-26T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "27",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Mar 27 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-27T00:00:00.000Z,
    },
    Object {
      "date": "28",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Mar 28 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-28T00:00:00.000Z,
    },
    Object {
      "date": "29",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Mar 29 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-29T00:00:00.000Z,
    },
    Object {
      "date": "30",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Mar 30 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-30T00:00:00.000Z,
    },
    Object {
      "date": "31",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Mar 31 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-03-31T00:00:00.000Z,
    },
    Object {
      "date": "1",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": true,
      "isToday": false,
      "key": "Fri Apr 01 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-01T00:00:00.000Z,
    },
    Object {
      "date": "2",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Apr 02 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-02T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "3",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Apr 03 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-03T00:00:00.000Z,
    },
    Object {
      "date": "4",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Apr 04 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-04T00:00:00.000Z,
    },
    Object {
      "date": "5",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Apr 05 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-05T00:00:00.000Z,
    },
    Object {
      "date": "6",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Apr 06 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-06T00:00:00.000Z,
    },
    Object {
      "date": "7",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Apr 07 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-07T00:00:00.000Z,
    },
    Object {
      "date": "8",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Fri Apr 08 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-08T00:00:00.000Z,
    },
    Object {
      "date": "9",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Apr 09 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-09T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "10",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Apr 10 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-10T00:00:00.000Z,
    },
    Object {
      "date": "11",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Apr 11 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-11T00:00:00.000Z,
    },
    Object {
      "date": "12",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Apr 12 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-12T00:00:00.000Z,
    },
    Object {
      "date": "13",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Apr 13 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-13T00:00:00.000Z,
    },
    Object {
      "date": "14",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Apr 14 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-14T00:00:00.000Z,
    },
    Object {
      "date": "15",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Fri Apr 15 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-15T00:00:00.000Z,
    },
    Object {
      "date": "16",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Apr 16 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-16T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "17",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Apr 17 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-17T00:00:00.000Z,
    },
    Object {
      "date": "18",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Apr 18 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-18T00:00:00.000Z,
    },
    Object {
      "date": "19",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Apr 19 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-19T00:00:00.000Z,
    },
    Object {
      "date": "20",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Apr 20 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-20T00:00:00.000Z,
    },
    Object {
      "date": "21",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Apr 21 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-21T00:00:00.000Z,
    },
    Object {
      "date": "22",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Fri Apr 22 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-22T00:00:00.000Z,
    },
    Object {
      "date": "23",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Apr 23 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-23T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "24",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sun Apr 24 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-24T00:00:00.000Z,
    },
    Object {
      "date": "25",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Mon Apr 25 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-25T00:00:00.000Z,
    },
    Object {
      "date": "26",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Tue Apr 26 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-26T00:00:00.000Z,
    },
    Object {
      "date": "27",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Wed Apr 27 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-27T00:00:00.000Z,
    },
    Object {
      "date": "28",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Thu Apr 28 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-28T00:00:00.000Z,
    },
    Object {
      "date": "29",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Fri Apr 29 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-29T00:00:00.000Z,
    },
    Object {
      "date": "30",
      "isInBounds": true,
      "isInMonth": true,
      "isSelected": false,
      "isToday": false,
      "key": "Sat Apr 30 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-04-30T00:00:00.000Z,
    },
  ],
  Array [
    Object {
      "date": "1",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Sun May 01 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-01T00:00:00.000Z,
    },
    Object {
      "date": "2",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Mon May 02 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-02T00:00:00.000Z,
    },
    Object {
      "date": "3",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Tue May 03 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-03T00:00:00.000Z,
    },
    Object {
      "date": "4",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Wed May 04 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-04T00:00:00.000Z,
    },
    Object {
      "date": "5",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Thu May 05 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-05T00:00:00.000Z,
    },
    Object {
      "date": "6",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Fri May 06 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-06T00:00:00.000Z,
    },
    Object {
      "date": "7",
      "isInBounds": true,
      "isInMonth": false,
      "isSelected": false,
      "isToday": false,
      "key": "Sat May 07 2016 00:00:00 GMT+0000 (Coordinated Universal Time)",
      "originalDate": 2016-05-07T00:00:00.000Z,
    },
  ],
]
`;
